### ISSUE TITLE
Allow custom Paint object to be used in renderToCanvas()

### ISSUE THREAD

**Issue by AdityaAnand1** [BigBadaboom/androidsvg#93]
I'm using this library with icon sets such as font awesome, material design icons etc. The SVGs of these (and most) icon sets use a single color.

Currently, if I want to give an icon a custom color: 

* I need to first allocate a bitmap (and it's canvas), then use renderToCanvas()
     
            Bitmap bitmap = Bitmap.createBitmap(size, size, Bitmap.Config.ARGB_8888);

            Canvas canvas = new Canvas(bitmap);

            svg.renderToCanvas(canvas);

            return bitmap;

* So now I have the icon on it's own dedicated bitmap. The color of the icon I get is the one used in the SVG file. Now the question comes, how do I set a custom color (without changing the SVG file)? Currently I use a very hacky changeBitmapColor() method that goes over every single pixel and literally alter the color of each pixel one by one. Say, if the icon is black, then find every black pixel and replace it with the custom color.

Clearly, this is less than ideal. 

* Firstly, I have to allocate a new bitmap specifically for using the icon
* The hacky method of changing the color is the opposite of performant
* Only after I've prepared the icon with the custom color, can I actually use that bitmap (by drawing to yet another canvas)

All of this could be fixed by having renderToCanvas() optionally take in a custom paint object (or simply a color) that overrides the SVG file color. (Like the official android canvas methods such as drawRect() or drawBitmap())

**Comment by BigBadaboom** [comment]
Thanks for your report.

There are probably quicker ways to achieve what you want rather than altering pixels yourself.  For instance you can set up a ColorFilter and use that when you paint your Bitmap.

For instance, try something like the following (not tested):

```
ColorMatrix colorMatrix = new ColorMatrix();
colorMatrix.setSaturation(0f);  // To B&W
ColorMatrixColorFilter colorFilter = new ColorMatrixColorFilter(colorMatrix);

Paint paint = new Paint();
paint.setColorFilter(colorFilter);   

canvas.drawBitmap(bitmap, 0, 0, paint);
```

However the main request to be able to modify the SVG before it is painted is already on the todo list as Issue #11.  So I am going to mark this as a duplicate of that one.

**Comment by AdityaAnand1** [comment]
Yes, that's exactly what I want to do but I can't because I only have:

```
 Canvas canvas = new Canvas(cachedBitmap);
 svg.renderToCanvas(canvas);
```

Nowhere to put a paint object (as in your snippet), hence no way to use ColorMatrix.

In a nutshell, I have single color SVGs, I'd like to override those colors programatically: (Right now, I have to choices, manually change pixel colors or edit the SVGs dynamically, by finding the color attributes and changing them in the string before passing them to `SVG.getFromString()`)

**Comment by BigBadaboom** [comment]
You should still be able to use my example.  It assumes you have already drawn the SVG onto a bitmap (as you are doing).

What are you doing with your `mBitmap` next?  If you are rendering it into another Canvas or bitmap, you can use the code above.  If you are sending it to an ImageView, you should be able to use `android:tint` etc.
